### PR TITLE
[ROCm] Add rocm support in nm.py lowering code

### PR DESCRIPTION
**Pull request by Ruturaj4**

### CODE REVIEW

**Comment by Ruturaj4** [comment]
@superbobry Could you please take a look?

File: jax/experimental/sparse/nm.py [hunk 1]
```diff
@@ -181,6 +181,9 @@ def _nm_spmm_abstract_eval(
 if gpu_sparse.cuda_is_supported:
   mlir.register_lowering(nm_spmm_p, _nm_spmm_gpu_lowering, platform="cuda")
 
+if gpu_sparse.rocm_is_supported:
+  mlir.register_lowering(nm_spmm_p, _nm_spmm_gpu_lowering, platform="ROCM")
```
Comment (hawkinsp) [inline jax/experimental/sparse/nm.py:185]: Nit: elsewhere we're using "rocm" (lower case), probably best to be consistent.

Comment (Ruturaj4) [reply jax/experimental/sparse/nm.py:185]: @hawkinsp Whenever we use `platform="ROCM"`. I observed we used caps. But, I can fix all of those if you would like

Comment (hawkinsp) [reply jax/experimental/sparse/nm.py:185]: Hmm. Interesting. No, I'd stick with "ROCM" then. I don't see any logic to canonicalize the platform name.

**Review by hawkinsp — APPROVED** [review]

File: jax/experimental/sparse/nm.py (hunk 1 above)
Comment (Ruturaj4) [reply jax/experimental/sparse/nm.py:185]: > Hmm. Interesting. No, I'd stick with "ROCM" then. I don't see any logic to canonicalize the platform name.

I also observed discrepancies in cuda. i.e. sometimes `platform="CUDA"` while other times platform="cuda"

**Pull request merged**